### PR TITLE
Fix shellcheck issues

### PR DESCRIPTION
**Pull request by asymmetric**

### CODE REVIEW

File: scripts/setup-env.sh [hunk 1]
```diff
@@ -1,9 +1,9 @@
 #!/usr/bin/env bash
-CHECK="test \"$_\" != \"$0\""
-$CHECK || { echo >&2 "Use this script by sourcing it \`. $0\` instead"; exit 1; }
+[[ "$_" != "$0" ]] || { echo >&2 "Use this script by sourcing it \`. $0\` instead"; exit 1; }
```
Comment (icetan) [inline scripts/setup-env.sh]: This doesn't work though.

Comment (asymmetric) [reply scripts/setup-env.sh]: There seem to be [better ways](https://stackoverflow.com/questions/2683279/how-to-detect-if-a-script-is-being-sourced#34642589) of doing it anyway, should we switch?

Comment (icetan) [reply scripts/setup-env.sh]: It does, sorry

Comment (icetan) [reply scripts/setup-env.sh]: It doesn't ran it with `sh scripts/setup-env.sh` and that didn't work before either

Comment (icetan) [reply scripts/setup-env.sh]: Cool hack using `return`, yeah much more robust.

**Pull request merged**